typespec-tests/src/main/java/com/cadl/patch/models/Resource.java

```diff
@@ -0,0 +1,355 @@
+        this.updatedProperties.add("intValue");
+        return this;
+    }
+
+    /**
+     * Get the enumValue property: The enumValue property.
+     * 
+     * @return the enumValue value.
+     */
+    @Generated
+    public ResourceEnumValue getEnumValue() {
+        return this.enumValue;
+    }
+
+    /**
+     * Set the enumValue property: The enumValue property.
+     * 
+     * @param enumValue the enumValue value to set.
+     * @return the Resource object itself.
+     */
+    @Generated
+    public Resource setEnumValue(ResourceEnumValue enumValue) {
+        this.enumValue = enumValue;
+        this.updatedProperties.add("enumValue");
+        return this;
+    }
+
+    /**
+     * Get the innerModelProperty property: The wireNameForInnerModelProperty property.
+     * 
+     * @return the innerModelProperty value.
+     */
+    @Generated
+    public InnerModel getInnerModelProperty() {
+        return this.innerModelProperty;
+    }
+
+    /**
+     * Set the innerModelProperty property: The wireNameForInnerModelProperty property.
+     * 
+     * @param innerModelProperty the innerModelProperty value to set.
+     * @return the Resource object itself.
+     */
+    @Generated
+    public Resource setInnerModelProperty(InnerModel innerModelProperty) {
+        this.innerModelProperty = innerModelProperty;
+        this.updatedProperties.add("innerModelProperty");
+        return this;
+    }
+
+    @Override
+    public JsonWriter toJson(JsonWriter jsonWriter) throws IOException {
+        if (jsonMergePatch) {
+            return toJsonMergePatch(jsonWriter);
+        } else {
+            jsonWriter.writeStartObject();
+            jsonWriter.writeMapField("map", this.map, (writer, element) -> writer.writeJson(element));
+            jsonWriter.writeStringField("description", this.description);
+            jsonWriter.writeNumberField("longValue", this.longValue);
+            jsonWriter.writeNumberField("intValue", this.intValue);
+            jsonWriter.writeStringField("enumValue", this.enumValue == null ? null : this.enumValue.toString());
+            jsonWriter.writeJsonField("wireNameForInnerModelProperty", this.innerModelProperty);
+            return jsonWriter.writeEndObject();
+        }
+    }
+
+    public JsonWriter toJsonMergePatch(JsonWriter jsonWriter) throws IOException {
+        jsonWriter.writeStartObject();
+        if (this.map != null) {
+            jsonWriter.writeMapField("map", this.map, (writer, element) -> {
+                element.serializeAsJsonMergePatch(true);
+                writer.writeJson(element);
+            });
+        } else if (updatedProperties.contains("map")) {
+            jsonWriter.writeNullField("map");
+        }
+        if (this.description != null) {
+            jsonWriter.writeStringField("description", this.description);
+        } else if (updatedProperties.contains("description")) {
+            jsonWriter.writeNullField("description");
+        }
```

Comment:
I am unsure, so just a discussion. Should we check `updatedProperties` before write any value (even if it is not null)?

Reply:
I think both are ok. Don't have preference actually.The current way seems straightforward to me.